src/codeql.ts

```diff
@@ -360,3 +353,27 @@ async function getRemoteQueryPackDefaultQuery(
 
   return queries[0];
 }
+
+export interface QueryRunMetadata {
```

Comment:
What do you think about moving all query run metadata code into its own module? It's currently spread between `codeql.ts` and `interpret.ts` and I'm thinking it doesn't have much to do with CodeQL.

We could have a `query-run-metadata.ts` that includes the `QueryRunMetadata` interface as well as `writeQueryRunMetadataToFile()` and `readQueryRunMetadataFromFile()` (currently called `readMetadata()`).

Note that if you make this change I'd suggest updating the `readQueryRunMetadataFromFile()` to receive a file path (the `response.downloadPath`) rather than the whole `DownloadResponse` object. That way the query run metadata module is agnostic of any response stuff.